en/concept/architecture.md

```diff
@@ -119,34 +119,53 @@ PX4 runs on various operating systems that provide a POSIX-API
 (such as Linux, macOS, NuttX or QuRT). It should also have some form of
 real-time scheduling (e.g. FIFO).
 
-The inter-module communication (using uORB) is based on shared memory. The whole
-PX4 middleware runs in a single address space, i.e. memory is shared between all
-modules. 
+The inter-module communication (using [uORB](../middleware/uorb.md)) is based on shared memory. 
+The whole PX4 middleware runs in a single address space, i.e. memory is shared between all modules. 
 
 > **Info** The system is designed such that with minimal effort it would
 > be possible to run each module in separate address space (parts that would need
 > to be changed include `uORB`, `parameter interface`, `dataman` and `perf`).
 
 There are 2 different ways that a module can be executed:
-- **Tasks**: this is the more common way. A module runs in its own task with its
-  own stack and process priority.
-- **Work queues**: the module runs on a shared task, meaning it does not own a
-  stack. Multiple tasks run on the same stack with a single priority per work
-  queue.
-  
+- **Tasks**: The module runs in its own task with its own stack and process priority
+  (this is the more common way). 
+- **Work queues**: The module runs on a shared task, meaning that it does not own a stack. 
+  Multiple tasks run on the same stack with a single priority per work queue.
+
   A task is scheduled by specifying a fixed time in the future.
   The advantage is that it uses less RAM, but the task is not allowed to sleep
   or poll on a message.
 
-  Work queues are used for periodic tasks, such as sensor drivers or the land
-  detector.
+  Work queues are used for periodic tasks, 
+  such as sensor drivers or the land detector.
 
 > **Note** Tasks running on a work queue do not show up in `top` 
 > (only the work queues themselves can be seen - e.g. as `lpwork`).
 
-The following sections provide additional OS-specific information.
 
-### NuttX
+### Background Programs/Daemons
```

Comment:
@bkueng Note that the terminology is not very "comfortable here". A daemon is commonly referred to as a background process, but we're also talking here about tasks, threads, and processes, where "process" has a different an distinct meaning, 
Anyway, I'm calling them background programs, which isn't perhaps all that great either.

Reply:
Yes, I can imagine that this section could be rather confusing. Maybe we should reduce it to just something like:
'`px4_task_spawn_cmd` is used to launch new tasks that run independently from the calling (parent) task'?
Plus the API example.

Reply:
Good idea. I've used your words and removed the words process and daemon (I'm OK with that because I don't think that is how people will search for the concept). See  https://github.com/PX4/Devguide/pull/396/commits/49ca6e2e78cdc2c7f9f3b2a7c380e921bffff2e2

You otherwise ok to merge this?